homeassistant/components/duotecno/switch.py

```diff
@@ -36,8 +37,14 @@ def is_on(self) -> bool:
 
     async def async_turn_on(self, **kwargs: Any) -> None:
         """Instruct the switch to turn on."""
-        await self._unit.turn_on()
+        try:
+            await self._unit.turn_on()
+        except Exception as err:  # pylint: disable=broad-except
```

Comment:
We should not catch broad exceptions here. Please make it more specific.

Reply:
I would recommend catching `OSError` in that case, this will include all I/O related failures.